internal/pkg/daemon/profilerecorder/profilerecorder.go

```diff
@@ -152,6 +184,9 @@ func (r *RecorderReconciler) isPodWithTraceAnnotation(obj runtime.Object) bool {
 
 // +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;watch
 func (r *RecorderReconciler) Reconcile(_ context.Context, req reconcile.Request) (reconcile.Result, error) {
+	// Mark the controller as ready if the first reconcile has been finished
+	defer func() { r.ready = true }()
```

Comment:
Do reconcile looks run serially? If they're concurrent, this could lead to a race condition

Reply:
Yeah, how about that: https://github.com/kubernetes-sigs/security-profiles-operator/blob/53651fe8313dcb84324f6e7c07ecd79ac40f9499/internal/pkg/atomic/bool.go